pkg/trie/triedb/node.go

```diff
@@ -168,8 +207,204 @@ func newNodeFromEncoded(nodeHash common.Hash, data []byte, storage NodeStorage)
 			children[i] = child
 		}
 
-		return Branch{partialKey: key, children: children, value: NewFromEncoded(value)}, nil
+		return Branch{partialKey: key, children: children, value: NewValueFromEncoded(encoded.PartialKey, value)}, nil
+	default:
+		panic("unreachable")
+	}
+}
+
+type NodeToEncode interface {
+	isNodeToEncode()
+}
+
+type (
+	NewNodeToEncode struct {
+		partialKey []byte
+		value      []byte
+	}
+	TrieNodeToEncode struct {
+		child NodeHandle
+	}
+)
+
+func (NewNodeToEncode) isNodeToEncode()  {}
+func (TrieNodeToEncode) isNodeToEncode() {}
+
+type ChildReference interface {
+	isChildReference()
```

Comment:
this method can be removed since `getNodeData` exists

Reply:
Removed, thks!